python/tank/api.py

```diff
@@ -489,11 +491,11 @@ def paths_from_template(self, template, fields, skip_keys=None, skip_missing_opt
         :rtype: List of strings.
         """
         skip_keys = skip_keys or []
-        if isinstance(skip_keys, basestring):
+        if isinstance(skip_keys, six.string_types):
             skip_keys = [skip_keys]
         
         # construct local fields dictionary that doesn't include any skip keys:
-        local_fields = dict((field, value) for field, value in fields.iteritems() if field not in skip_keys)
+        local_fields = dict((field, value) for field, value in six.iteritems(fields) if field not in skip_keys)
```

Comment:
Once again, we don't have that many items, so we could simply call `items()` to keep the code simple or do something like
```suggestion
        local_fields = dict((field, fields[field]) for field in fields if field not in skip_keys)
```

Reply:
✅